src/view/com/util/post-ctrls/RepostButton.tsx

```diff
@@ -1,108 +1,129 @@
-        style={
+    <>
+      <Button
+        testID="repostBtn"
+        onPress={() => {
+          requireAuth(() => dialogControl.open())
+        }}
+        style={[a.flex_row, a.align_center, a.gap_xs, {padding: 5}]}
+        label={`${
           isReposted
-            ? (styles.reposted as StyleProp<ViewStyle>)
-            : defaultControlColor
-        }
-        strokeWidth={2.4}
-        size={big ? 24 : 20}
-      />
-      {typeof repostCount !== 'undefined' && repostCount > 0 ? (
-        <Text
-          testID="repostCount"
-          style={
-            isReposted
-              ? [s.bold, s.green3, s.f15, s.ml5]
-              : [defaultControlColor, s.f15, s.ml5]
-          }>
-          {repostCount}
-        </Text>
-      ) : undefined}
-    </TouchableOpacity>
+            ? _(msg`Undo repost`)
+            : _(msg({message: 'Repost', context: 'action'}))
+        } (${repostCount} ${pluralize(repostCount || 0, 'repost')})`}
+        shape="round"
+        variant="ghost"
+        color="secondary">
+        <Repost style={color} size={size} />
+        {typeof repostCount !== 'undefined' && repostCount > 0 ? (
+          <Text
+            testID="repostCount"
+            style={[color, a.text_md, isReposted && a.font_bold]}>
+            {repostCount}
+          </Text>
+        ) : undefined}
+      </Button>
+      <Dialog.Outer control={dialogControl}>
+        <Dialog.Handle />
+        <Dialog.Inner label={_(msg`Repost or quote post`)}>
+          <View style={a.gap_xl}>
+            <View style={a.gap_xs}>
+              <Button
+                style={[a.justify_start, a.px_sm]}
+                label={
+                  isReposted
+                    ? _(msg`Remove repost`)
+                    : _(msg({message: `Repost`, context: 'action'}))
+                }
+                onPress={() => {
+                  dialogControl.close()
+                  onRepost()
+                }}
+                size="large"
+                variant="outline">
+                <ButtonIcon
+                  icon={Repost}
+                  size="lg"
+                  color={t.palette.primary_500}
+                />
+                <ButtonText style={a.text_xl}>
+                  {isReposted
+                    ? _(msg`Remove repost`)
+                    : _(msg({message: `Repost`, context: 'action'}))}
+                </ButtonText>
+              </Button>
+              <Button
+                style={[a.justify_start, a.px_sm]}
+                label={_(msg`Quote post`)}
+                onPress={() => {
+                  dialogControl.close()
+                  onQuote()
+                }}
+                size="large"
+                variant="outline">
+                <ButtonIcon
+                  icon={Quote}
+                  size="lg"
+                  color={t.palette.primary_500}
+                />
+                <ButtonText style={a.text_xl}>{_(msg`Quote post`)}</ButtonText>
```

Comment:
These two components are really just for consistency when creating "normal" buttons. For custom stuff like this I don't think there's any benefit from using them (and extending their APIs to support stuff like `color`).